### ISSUE TITLE
[Backend] Build secret endpoint to trigger async jobs

### ISSUE THREAD

**Issue by sunil-sadasivan** [department-of-veterans-affairs/caseflow#3179]
**Description**

We need a secret endpoint to trigger jobs asynchronously.  This will be used by lambda to run on a cron schedule.  

**Testing**

- [ ] Create a new API key on each environment.  
- [ ] Run the heartbeat job.  (Replace token and host with the right environment details)

FIXED route details:
```
curl localhost:3000/api/v1/jobs/start_async -d  'job_type=heartbeat' -H "Authorization: Token token=PUBLICDEMO123"
```

**Comment by sunil-sadasivan** [comment]
PR: https://github.com/department-of-veterans-affairs/caseflow/pull/3162

**Comment by sunil-sadasivan** [comment]
**AC:**

This endpoint allows for background jobs (heartbeat, create_establish_claim, prepare_establish_claim) to be triggered.  This will be used for lambda to trigger these on a schedule. 

**To Test:**

Generate a token for each environment:

```
key=ApiKey.create!(consumer_name: "jobs_trigger")
key.key_string
```
This key_string should be recorded in OnePassword. 

```
curl https://certification.cf.uat.ds.va.gov/jobs/start_async -d 'job_type=heartbeat' -H "Authorization: Token token=<TOKEN>"
```
Proper output should be `{"success":true,"job_id":"e75c8ef3-9495-4409-8252-63124a360a11"}`.  

If a bad token is provided: `{"status":"unauthorized"}`

**Comment by kierachell** [comment]
Seems like authentication is not quite setup properly.  Check `api/v1` for how it's bypassed, otherwise I am being redirected to `/login` page or `404`

**Comment by sunil-sadasivan** [comment]
Fixing with #3207

**Comment by sunil-sadasivan** [comment]
@kierachell, merged adjustments with #3207 

New route details:
```
curl localhost:3000/api/v1/jobs/start_async -d  'job_type=heartbeat' -H "Authorization: Token token=PUBLICDEMO123"
```

**Comment by kierachell** [comment]
**PASSED**
`curl -v -d 'job_type=prepare_establish_claim' -H "Authorization: Token token=******" https://appeals.cf.uat.ds.va.gov/api/v1/jobs`

Make sure it's not `start_async`

<img width="506" alt="screen shot 2017-09-12 at 9 01 44 am" src="https://user-images.githubusercontent.com/18075411/30327146-04866f1e-9799-11e7-88c8-a045d97d102c.png">

```
{"success":true,"job_id":"078273c7-502b-47c7-ba42-f277308703ae"}
```
